### PR TITLE
[react-bootstrap-table-next] Add typing for react-bootstrap-table-next

### PR DESCRIPTION
**Pull request by gosticks**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [x] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

Adding typings for **react-bootstrap-table-next** and its additional modules:
- **react-bootstrap-table2-filter**
- **react-bootstrap-table2-pagination**
- **react-bootstrap-table2-toolkit**

~~Unfortunately I could not run `npm run lint` on my system and changing the module resolution to node did not fix this issue for me. _(Error message: TS2307: Cannot find module 'csstype' or its corresponding type declarations)_ 
To my understanding this is a problem on my machine and not related to the typings as running lint on other type definitions using `React.CSSProperties` also results in this error.~~

### CODE REVIEW

**Review by uniqueiniquity — CHANGES_REQUESTED** [review]

**Comment by gosticks** [comment]
I've changed the minimum typescript version in all packages. Now all except for `react-bootstrap-table2-filter` require version TS 3.0. Strangely the filter package would not pass the tests with a version below TS 3.7.  Therefore I compared the new features in 3.7 with the filter code and could not find a reason why it fails the tests.

File: types/react-bootstrap-table2-filter/react-bootstrap-table2-filter-tests.tsx [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+);
+
+/**
+ * Single select column test
+ */
+const selectColumns = [
+    {
+        dataField: 'quality',
+        text: 'Product Quailty',
+        formatter: (cell: number) => selectOptionsList.find(val => val.value === cell)?.value ?? '',
```
Comment (uniqueiniquity) [inline types/react-bootstrap-table2-filter/react-bootstrap-table2-filter-tests.tsx:214]: It's the optional chaining (`?.`) in this line that's preventing it from working on versions earlier than 3.7

**Review by uniqueiniquity — CHANGES_REQUESTED** [review]

File: types/react-bootstrap-table2-filter/react-bootstrap-table2-filter-tests.tsx (hunk 1 above)
Comment (gosticks) [reply types/react-bootstrap-table2-filter/react-bootstrap-table2-filter-tests.tsx]: Oh sorry, I don't know how I could have missed that. I just fixed that and updated the TS version to 3.0.

**Review by uniqueiniquity — APPROVED** [review]

**Comment by uniqueiniquity** [comment]
Congratulations on your first DefinitelyTyped contribution!
Thank you for being a part of the community!

**Pull request merged**